modular_zubbers/master_files/code/modules/loadout/loadout_preference.dm

```diff
@@ -0,0 +1,39 @@
+// Oh, I'm sorry, you were looking for GOOD code? Turn around and leave. - Rimi
+
+/datum/preference/loadout_index
+	savefile_key = "loadout_index"
+	savefile_identifier = PREFERENCE_CHARACTER
+	can_randomize = FALSE
+
+/datum/preference/loadout_index/apply_to_human(mob/living/carbon/human/target, value, datum/preferences/preferences)
+	return
+
+/datum/preference/loadout_index/create_informed_default_value(datum/preferences/preferences)
+	var/list/loadouts = preferences.read_preference(/datum/preference/loadout)
+	// Explanation so I don't get more questions:
+	// You can't get a list of keys for assoc lists, so you have to iterate said list to get them.
+	// I only need the fist one, so I just immediately return on the first loop.
+	for (var/name in loadouts) // Lazy hack fraud
+		return name
```

Comment:
```suggestion
	if(length(loadouts))
		return loadouts[1]
```
This is a bit faster and does the same thing without being hacky.

Snippet I used to make sure this is the case:
```
/proc/list_finder_run()
	var/list/L = list("a" = "a1","b" = "b1","c" = "c1")
	var/i
	for(i=0, i<=100000, i++)
		var/a = list_finder_ifcheck(L)
		var/b = list_finder_forcheck(L)

/proc/list_finder_ifcheck(var/list/L)
	if(length(L))
		return L[1]

/proc/list_finder_forcheck(var/list/L)
	for(var/name in L)
		return name
```

Reply:
I love undocumented functionality!